src/plugins/formulas/formulas.js

```diff
@@ -203,36 +250,231 @@ export class Formulas extends BasePlugin {
       });
     }
 
+    const hfSetUpProperly = this.setupHF();
+
+    if (!hfSetUpProperly) {
+      return;
+    }
+
+    // HyperFormula events:
+    this.hyperformula.on('valuesUpdated', (...args) => this.onHFvaluesUpdated(...args));
+
+    this.applyHFSettings();
+
     super.enablePlugin();
   }
 
   /**
    * Disables the plugin functionality for this Handsontable instance.
    */
   disablePlugin() {
-    // TODO add tests for this line
-    this.hyperformula.destroy();
-
     super.disablePlugin();
   }
 
+  /**
+   * Triggered on `updateSettings`.
+   */
+  updatePlugin() {
+    this.#settings = this.hot.getSettings()[PLUGIN_KEY];
+
+    this.applyHFSettings();
+
+    if (isDefined(this.#settings.sheetName) && this.#settings.sheetName !== this.sheetName) {
+      this.switchSheet(this.#settings.sheetName);
+    }
+
+    super.updatePlugin();
+  }
+
   /**
    * Destroys the plugin instance.
    */
   destroy() {
+    if (this.hyperformula) {
+      const hfInstances = staticRegister('formulas').getItem('hyperformulaInstances');
+      const sharedHFInstanceUsage = hfInstances ? hfInstances.get(this.hyperformula) : null;
+
+      if (sharedHFInstanceUsage && sharedHFInstanceUsage.includes(this.hot.guid)) {
+        sharedHFInstanceUsage.splice(
+          sharedHFInstanceUsage.indexOf(this.hot.guid),
+          1
+        );
+
+        if (sharedHFInstanceUsage.length === 0) {
+          hfInstances.delete(this.hyperformula);
+          this.hyperformula.destroy();
+        }
+      }
+    }
+
     super.destroy();
   }
 
+  /**
+   * Add a sheet to the shared HyperFormula instance.
+   *
+   * @param {Array} sheetData Data passed to the shared HyperFormula instance. Has to be declared as an array of
+   * arrays - array of objects is not supported in this scenario.
+   * @param {string} [sheetName] The new sheet name. If not provided, will be auto-generated by HyperFormula.
+   * @param {boolean} [autoLoad] If `true`, the new sheet will be loaded into the Handsontable instance.
+   * @returns {boolean} `false` if the data format is unusable, `true` otherwise.
+   */
+  addSheet(sheetData, sheetName, autoLoad) {
+    if (
+      !sheetData ||
+      !Array.isArray(sheetData) ||
+      (sheetData.length && !Array.isArray(sheetData[0]))
+    ) {
+      error('The provided data should be an array of arrays.');
```

Comment:
Why not `throw`?

Reply:
Because throwing an actual error would cause the table to crash, and this is more like a more severe warning kind of thing, and the table keeps on working fine. I _think_ that's better in this case, WDYT?

Reply:
Oh, in that case I don't mind.